models/rubicon/arbitrum/rubicon_arbitrum_offers.sql

```diff
@@ -0,0 +1,215 @@
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index'],
+    post_hook='{{ expose_spells(\'["arbitrum"]\',
+                                "project",
+                                "rubicon",
+                                \'["denver"]\') }}'
+    )
+}}
+
+{% set project_start_date = '2023-06-09' %}
+   
+WITH offers AS 
+(
+    -- select the offer data from the emitOffer event
+    SELECT 
+        e.evt_block_time AS block_time,
+        e.evt_block_number AS block_number,
+        e.pay_gem AS sell_token_address,
+        e.buy_gem AS buy_token_address,
+        e.pay_amt AS sell_amount_raw,
+        e.buy_amt AS buy_amount_raw,
+        e.contract_address AS project_contract_address,
+        e.evt_tx_hash AS tx_hash,
+        e.evt_index,
+        e.maker,
+        e.id AS offer_id
+    FROM {{ source('rubicon_arbitrum', 'RubiconMarket_evt_emitOffer') }} e
+
+    -- filter out offers that were created before the project start date
+    WHERE e.evt_block_time >= cast('{{ project_start_date }}' AS timestamp)
+    {% if is_incremental() %} -- only run this filter if it is an incremental run
+    AND e.evt_block_time >= date_trunc('day', now() - interval '1 week')
+    {% endif %}
+),
+
+trades AS 
+(
+    -- select the trade data from the emitTake event
+    SELECT
+        t.id AS offer_id,
+        t.pay_gem AS sell_token_address,
+        t.buy_gem AS buy_token_address,
+        CAST(t.take_amt AS DECIMAL(38,0)) AS sold_amount_raw,
+        CAST(t.give_amt AS DECIMAL(38,0)) AS bought_amount_raw,
+        erc20_sell.symbol AS sell_token_symbol,
+        erc20_buy.symbol AS buy_token_symbol,
+        CAST(t.take_amt AS DECIMAL(38,0)) / power(10, erc20_sell.decimals) AS sold_amount,
+        CAST(t.give_amt AS DECIMAL(38,0)) / power(10, erc20_buy.decimals) AS bought_amount,
+        (CAST(t.take_amt AS DECIMAL(38,0)) / power(10, erc20_sell.decimals)) * sell_token_price.price AS sold_amount_usd,
+        (CAST(t.give_amt AS DECIMAL(38,0)) / power(10, erc20_buy.decimals)) * buy_token_price.price AS bought_amount_usd
+    FROM {{ source('rubicon_arbitrum', 'RubiconMarket_evt_emitTake') }} t 
+
+    -- get the relevant sell token data
+    LEFT JOIN {{ ref('tokens_erc20') }} erc20_sell
+        ON erc20_sell.contract_address = t.pay_gem
+        AND erc20_sell.blockchain = 'arbitrum'
+
+    -- get the relevant buy token data
+    LEFT JOIN {{ ref('tokens_erc20') }} erc20_buy
+        ON erc20_buy.contract_address = t.buy_gem
+        AND erc20_buy.blockchain = 'arbitrum'
+
+    -- get the sell token price
+    LEFT JOIN {{ source('prices', 'usd') }} sell_token_price
+        ON sell_token_price.minute = date_trunc('minute', t.evt_block_time)
+        AND sell_token_price.contract_address = t.pay_gem
+        AND sell_token_price.blockchain = 'arbitrum'
+        {% if not is_incremental() %} -- only run this filter if it is an incremental run
```

Comment:
you can also include an `else` here and do the incremental filter, similar to decoded source tables